src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,56 @@
+.calendar {
+  display: flex;
+  flex-direction: row;
+  flex-wrap: wrap;
+  padding: 10px;
+  gap: $daysGap;
+  width: calc(7*100px + 7*1px);
+
+  &__day {
+    box-sizing: border-box;
+    background-color: #eee;
+    width: $dimensions;
+    height: $dimensions;
+    text-align: center;
+    line-height: $dimensions;
+    border: 1px solid black;
+    transition: transform 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: pink;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $day from 1 through 31 {
+    &__day--#{$day}::before {
+      content: '#{$day}';
+      font-size: 30px;
+      font-family: Arial, Helvetica, sans-serif;
+    }
+  }
+
+  $days: (
+    "mon": 0,
+    "tue": calc(1*#{$dimensions} + 1*#{$daysGap}),
+    "wed": calc(2*#{$dimensions} + 2*#{$daysGap}),
+    "thu": calc(3*#{$dimensions} + 3*#{$daysGap}),
+    "fri": calc(4*#{$dimensions} + 4*#{$daysGap}),
+    "sat": calc(5*#{$dimensions} + 5*#{$daysGap}),
+    "sun": calc(6*#{$dimensions} + 6*#{$daysGap})
```

Comment:
Mayber instead of calculating the margin for each day here, simply put the index of each day and calculate it inside @each with help of other variables.

Btw, this could be inside of variables file